conf/scalardb-custom-values.yaml

```diff
@@ -17,34 +17,22 @@ envoy:
       service.beta.kubernetes.io/aws-load-balancer-internal: "true"
       service.beta.kubernetes.io/aws-load-balancer-type: "nlb"
 
-  tolerations:
-    - effect: NoSchedule
-      key: scalar-labs.com/dedicated-node
-      operator: Equal
-      value: scalardb
-
   affinity:
-    nodeAffinity:
-      requiredDuringSchedulingIgnoredDuringExecution:
-        nodeSelectorTerms:
-          - matchExpressions:
-              - key: scalar-labs.com/dedicated-node
-                operator: In
-                values:
-                  - scalardb
     podAntiAffinity:
-      requiredDuringSchedulingIgnoredDuringExecution:
-        - labelSelector:
-            matchExpressions:
-              - key: app.kubernetes.io/name
-                operator: In
-                values:
-                  - scalardb
-              - key: app.kubernetes.io/app
-                operator: In
-                values:
-                  - envoy
-          topologyKey: kubernetes.io/hostname
+      preferredDuringSchedulingIgnoredDuringExecution:
+        - podAffinityTerm:
+            labelSelector:
+              matchExpressions:
+                - key: app.kubernetes.io/name
+                  operator: In
+                  values:
+                    - scalardb
+                - key: app.kubernetes.io/app
+                  operator: In
+                  values:
+                    - envoy
+            topologyKey: kubernetes.io/hostname
+          weight: 50
```

Comment:
@komamitsu  
Thank you for your review and question!

Actually, this value has almost no meaning in this sample configuration.

The `weight` value is used when Kubernetes decides the priority if there are two or more affinity configurations. You can see the sample configuration that has two affinity configurations in the following official document.
https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#node-affinity-weight

As the above, the `weight` has meaning when there are two or more affinity configurations, but it has almost no meaning if there is only one affinity configuration.

However, this value is the `required` parameter as follows.

```console
$ kubectl explain pod.spec.affinity.podAntiAffinity.preferredDuringSchedulingIgnoredDuringExecution
KIND:       Pod
VERSION:    v1

FIELD: preferredDuringSchedulingIgnoredDuringExecution <[]WeightedPodAffinityTerm>

DESCRIPTION:
    The scheduler will prefer to schedule pods to nodes that satisfy the
    anti-affinity expressions specified by this field, but it may choose a node
    that violates one or more of the expressions. The node that is most
    preferred is the one with the greatest sum of weights, i.e. for each node
    that meets all of the scheduling requirements (resource request,
    requiredDuringScheduling anti-affinity expressions, etc.), compute a sum by
    iterating through the elements of this field and adding "weight" to the sum
    if the node has pods which matches the corresponding podAffinityTerm; the
    node(s) with the highest sum are the most preferred.
    The weights of all of the matched WeightedPodAffinityTerm fields are added
    per-node to find the most preferred node(s)

FIELDS:
  podAffinityTerm       <PodAffinityTerm> -required-
    Required. A pod affinity term, associated with the corresponding weight.

  weight        <integer> -required-
    weight associated with matching the corresponding podAffinityTerm, in the
    range 1-100.
```

FYI: You also can see the same information in the following API reference.
https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#PodAntiAffinity

If we don't set `wight`, we will face the following error and you cannot deploy pods.
```console
$ helm install scalardb-cluster scalar-labs/scalardb-cluster -f ./scalardb-cluster.yaml
Error: INSTALLATION FAILED: Deployment.apps "scalardb-cluster-node" is invalid: spec.template.spec.affinity.podAntiAffinity.preferredDuringSchedulingIgnoredDuringExecution[0].weight: Invalid value: 0: must be in the range 1-100
```

So, we have to set this value even if there is only one affinity configuration. But, as I mentioned, it has almost no meaning.

Therefore, I just set the middle value `50` (this parameter can accept the range `1-100`). This is why the value is `50`. Actually, there is no strong reason for this value.

Reply:
@kota2and3kan Thanks for the explanation! Understood.